### ISSUE TITLE
Question about conditioning f0 on MIDI pitches

### ISSUE THREAD

**Issue by faniuy** [magenta/ddsp#67]
Since MIDI pitches are essentially correspoding to the base frequencies of instrumental notes, it is intuitive to condition f0 on MIDI pitches. And I even see that you mentioned the MIDI data on Nsynth dataset. However, I never found the usage of MIDI data. So,
1. Have you ever tried to condition f0 on MIDI pitches?
2. Are there any particular reason that prevents to do so? (e.g. maybe an instrumental note is not always in the same pitch in its duration, and the model relies on the exact pitch?)

**Comment by jesseengel** [comment]
It should work to some degree. True instrument pitch often has small variations over time, not to mention bends and slides and what not. So it will make sound, but it might sound a bit synthetic. We am currently working on some models that learn to predict true pitch variations from a MIDI signal.

**Comment by faniuy** [comment]
Sounds great!
I think it's even possible to deal with polyphonic sounds if the encoder doesn't rely on extracting pitch from wav.
I'll try it out too. Thank you!

**Comment by erl-j** [comment]
This is the actually subject of my master thesis. I've posted some audio examples on this [website](https://erl-j.github.io/SSSWebsite/) (still under construction) . In short, I trained a separate model to predict pitch and loudness from midi like @jesseengel mentions above. I then fed the predicted pitch and loudness features into a pretrained ddsp autoencoder. There is lots of room for improvement in my implementation but the overall method seems very promising.
